frozen.py

```diff
@@ -1,54 +1,65 @@
 import gym
 import numpy as np
+import argparse
 
 env = gym.make('FrozenLake-v0')
 env.reset()
 
-NUM_ACTIONS=env.action_space.n
-NUM_STATES= env.observation_space.n
-q_table=np.zeros((NUM_STATES, NUM_ACTIONS))
+NUM_ACTIONS = env.action_space.n
+NUM_STATES = env.observation_space.n
 """print q_table"""
 
-
 # Q-Learning Parameters
 
-G=.99
-L=.75
-
-rL= []
-
-
-#Learning
-episodes=1500
-
-for i_episode in range(episodes):
-	state= env.reset()
-	
-	rall=0
-
-	for t in range(100):
-	    env.render()
-	    # action = env.action_space.sample()
-	    action=np.argmax(q_table[state,:] + np.random.randn(1,NUM_ACTIONS) * 1./(i_episode+1))
-	    
-
-	    next_state, reward, done, info = env.step(action)
-
-	    rall+=reward
-	    # update Q table
-	    q_table[state,action]+= L * ( reward + G * (np.max(q_table[next_state,:]))-q_table[state,action])
-
-	    state=next_state
-	    if done:
-	        print("Episode finished after {} timesteps".format(t+1))
-	        break
-	rL.append(rall)
-
-
-
-print "score" + str(sum(rL)/episodes)
-print "G:%f L:%f" %(G,L)
-print "Resulting Q Table"
-print q_table
-
+def run_episodes(G, learning_rate, num_episodes):
+    rL= []
+    q_table=np.zeros((NUM_STATES, NUM_ACTIONS))
+
+    for i_episode in range(num_episodes):
+        state = env.reset()
+        rall = 0
+
+        for t in range(100):
+            env.render()
+            # action = env.action_space.sample()
+            action=np.argmax(q_table[state,:] + np.random.randn(1,NUM_ACTIONS) * 1./(i_episode+1))
+            next_state, reward, done, info = env.step(action)
+            rall += reward
+
+            # update Q table
+            q_table[state, action] += learning_rate * (reward + G * (np.max(q_table[next_state, :])) - q_table[state,action])
+            state = next_state
+
+            if done:
+                print("Episode finished after {} timesteps".format(t + 1))
+                break
+        rL.append(rall)
+
+    print "score" + str( sum(rL) / num_episodes)
+    print "G:%f L:%f" % (G, learning_rate)
+    print "Resulting Q Table"
+    print q_table
+
+if __name__ == '__main__':
+    """
+    take input from the command line and run your game
+    @see https://docs.python.org/3/library/argparse.html
+    for more examples
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('g', type=float,
+                        default=1.0, help='gamma')
+    parser.add_argument('l', type=float,
+                        default=0.5, help='learning rate')
+    parser.add_argument('e', type=int, default=1500,
+                        help='number of episodes')
+    args = parser.parse_args()
```

Comment:
very common python module to get arguments from command line easily.
if you uncomment this, you can call the program like,
`python frozen.py 0.5 0.5 1500` and it will run the simulation with `gamma = 0.5, learning_rate = 0.5`, and `num_episodes = 1500`